frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx

```diff
@@ -0,0 +1,383 @@
+
+    const [_, estimated_texture_size] = selectMagForTextureCreation(
+      colorLayer,
+      selectedBoundingBox,
+    );
+
+    const hasEnoughMags = estimated_texture_size < 1.5 * TARGET_TEXTURE_SIZE;
+    if (hasEnoughMags)
+      errorMessages.push(
+        "The selected bounding box is too large to create an animation. Either shrink the bounding box or consider downsampling the dataset to coarser magnifications.",
+      );
+
+    const isDtypeSupported = colorLayer.elementClass !== "uint24";
+    if (isDtypeSupported)
+      errorMessages.push("Sorry, animations are not supported for uInt24 datasets.");
+
+    const isDataset3D =
+      !is2dDataset(state.dataset) && (colorLayer.additionalAxes?.length || 0) === 0;
+    if (isDataset3D) errorMessages.push("Sorry, animations are only supported for 3D datasets.");
+
+    const isTooManyMeshes = meshSegmentIds.length > 30;
+    if (isTooManyMeshes)
+      errorMessages.push(
+        "You selected too many meshes for the animation. Please keep the number of meshes below 30 to create an animation.",
+      );
+
+    const validationStatus = hasEnoughMags && isDtypeSupported && isDataset3D && !isTooManyMeshes;
+
+    setValidationErrors(errorMessages);
+    setIsValid(validationStatus);
+
+    return validationStatus;
+  };
+
+  const submitJob = (evt: React.MouseEvent) => {
+    const state = Store.getState();
+    const boundingBox = userBoundingBoxes.find(
+      (bb) => bb.id === selectedBoundingBoxId,
+    )!.boundingBox;
+
+    // Submit currently visible pre-computed meshes
+    let meshSegmentIds = [] as number[];
+    let meshFileName = "meshfile.hdf5";
+    let segmentationLayerName = "segmentatation";
+
+    const visibleSegmentationLayer = Model.getVisibleSegmentationLayer();
+
+    if (visibleSegmentationLayer) {
+      const availableMeshes = state.localSegmentationData[visibleSegmentationLayer.name].meshes;
+      meshSegmentIds = Object.values(availableMeshes)
+        .filter((mesh) => mesh.isVisible && mesh.isPrecomputed)
+        .map((mesh) => mesh.segmentId);
+
+      const currenMeshFile =
+        state.localSegmentationData[visibleSegmentationLayer.name].currentMeshFile;
+      meshFileName = currenMeshFile?.meshFileName || meshFileName;
+
+      if (visibleSegmentationLayer.fallbackLayerInfo) {
+        segmentationLayerName = visibleSegmentationLayer.fallbackLayerInfo.name;
+      } else {
+        segmentationLayerName = visibleSegmentationLayer.name;
+      }
+    }
+
+    // Submit the configured min/max intensity info to support float datasets
+    const selectedColorLayer = getLayerByName(dataset, selectedColorLayerName);
+    const layerConfiguration = state.datasetConfiguration.layers[selectedColorLayer.name];
+    const { intensityRange } = layerConfiguration;
+    const defaultIntensityRange = getDefaultValueRangeOfLayer(dataset, selectedColorLayer.name);
+    const [intensityMin, intensityMax] = intensityRange || defaultIntensityRange;
```

Comment:
can you refactor this into a `getEffectiveIntensityRange(dataset, layerName, datasetConfiguration)` auxiliary accessor. Could be added to dataset_accessors.ts.